shtdlib.sh

```diff
@@ -1996,13 +1996,44 @@ function ln_sf {
 # Create string of random characters
 #  - First param is length, default: 20
 #  - Second param is characters, default: A-Za-z0-9_ (Note: '-' specifies range)
-gen_rand_chars() {
+function gen_rand_chars {
+    shopt_decorator_option_name='pipefail'
+    shopt_decorator_option_value='false'
+    # shellcheck disable=2015
+    shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return || conditional_exit_on_fail 121 "Failed to run ${FUNCNAME[0]} with shopt_decorator"
+
     local length="${1:-20}"
     local chars="${2:-A-Za-z0-9_}"
     debug 10 "Creating a string of random characters of length: ${length} and chars: ${chars}"
     LC_CTYPE=C tr -dc "${chars}" < '/dev/urandom' | head -c "${length}"
 }
 
+# Checks if an environtment variable is set and contains a string longer than
+# 0, if not then it's set to a random value.
+# If a file name/path is specified then a line containing VARIABLE=VALUE is
+# written to the end of the file. Optionally the length of the random
+# string/value can be specified. (defaults to 50)
+function auto_ensure_key_exists {
```

Comment:
Would `ensure_pass_key_exists` make more sense as a name for this function? It seems to be more for checking if a password value exists then just a key value.

Reply:
It's not just for passwords, also for random strings and key/value pairs.

How about: check_set_persist_random_variable ?

Reply:
Sure that sounds a little more clear to me.